pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotTaskManager.java

```diff
@@ -62,6 +89,220 @@ public PinotTaskManager(PinotHelixTaskResourceManager helixTaskResourceManager,
+            tableToDelete.removeAll(currentChilds);
+            if (!tableToDelete.isEmpty()) {
+              LOGGER.info("Found tables to clean up cron task scheduler: {}", tableToDelete);
+              for (String tableWithType : tableToDelete) {
+                cleanUpCronTaskSchedulerForTable(tableWithType);
+              }
+            }
+          });
+      for (String tableWithType : helixResourceManager.getAllTables()) {
+        subscribeTableConfigChanges(tableWithType);
+      }
+    } catch (SchedulerException e) {
+      LOGGER.error("Unable to create a scheduler.", e);
+      _scheduledExecutorService = null;
+    }
+  }
+
+  private String getPropertyStorePathForTable(String tableWithType) {
+    return TABLE_CONFIG_PATH_PREFIX + tableWithType;
+  }
+
+  protected synchronized void cleanupCronTaskScheduler() {
+    try {
+      _scheduledExecutorService.clear();
+    } catch (SchedulerException e) {
+      LOGGER.error("Failed to clear all tasks in scheduler", e);
+    }
+  }
+
+  public synchronized void cleanUpCronTaskSchedulerForTable(String tableWithType) {
+    LOGGER.info("Cleaning up task in scheduler for table {}", tableWithType);
+    TableTaskSchedulerUpdater tableTaskSchedulerUpdater = _tableTaskSchedulerUpdaterMap.get(tableWithType);
+    _pinotHelixResourceManager.getPropertyStore()
+        .unsubscribeDataChanges(getPropertyStorePathForTable(tableWithType), tableTaskSchedulerUpdater);
+    removeAllTasksFromCronExpressions(tableWithType);
+    _tableTaskSchedulerUpdaterMap.remove(tableWithType);
+  }
+
+  private synchronized void removeAllTasksFromCronExpressions(String tableWithType) {
+    Set<JobKey> jobKeys;
+    try {
+      jobKeys = _scheduledExecutorService.getJobKeys(GroupMatcher.anyJobGroup());
+    } catch (SchedulerException e) {
+      LOGGER.error("Got exception when fetching all jobKeys", e);
+      return;
+    }
+    for (JobKey jobKey : jobKeys) {
+      if (jobKey.getName().equalsIgnoreCase(tableWithType)) {
+        try {
+          _scheduledExecutorService.deleteJob(jobKey);
+        } catch (SchedulerException e) {
+          LOGGER.error("Got exception when deleting the scheduled job - {}", jobKey, e);
+        }
+      }
+    }
+    _tableTaskTypeToCronExpressionMap.remove(tableWithType);
+  }
+
+  public synchronized void subscribeTableConfigChanges(String tableWithType) {
+    if (_tableTaskSchedulerUpdaterMap.containsKey(tableWithType)) {
+      return;
+    }
+    TableTaskSchedulerUpdater tableTaskSchedulerUpdater = new TableTaskSchedulerUpdater(tableWithType, this);
+    _pinotHelixResourceManager.getPropertyStore()
+        .subscribeDataChanges(getPropertyStorePathForTable(tableWithType), tableTaskSchedulerUpdater);
+    _tableTaskSchedulerUpdaterMap.put(tableWithType, tableTaskSchedulerUpdater);
+    LOGGER.info("Trying to update task schedule for table: {}", tableWithType);
+    try {
+      updateCronTaskScheduler(tableWithType);
```

Comment:
This is redundant. The callback should be triggered when subscribing the path

Reply:
From the test, this is not expected behavior. I need to call this explicitly for the first time.